### PR TITLE
buf: 1.0.0-rc11 -> 1.0.0-rc12

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Motivation for this change

https://github.com/bufbuild/buf/releases/tag/v1.0.0-rc12

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/buf/default.nix [hunk 1]
```diff
@@ -26,11 +27,12 @@ buildGoModule rec {
     ./skip_test_requiring_dotgit.patch
   ];
 
-  nativeBuildInputs = [ protobuf ];
+  nativeBuildInputs = [ protobuf installShellFiles ];
   # Required for TestGitCloner
   checkInputs = [ git ];
 
   ldflags = [ "-s" "-w" ];
+  tags = [ "-trimpath" ];
```
Comment (lrewega) [inline pkgs/development/tools/buf/default.nix:35]: I believe `-trimpath` is the default for `buildGoModule`, and also this isn't a valid build tag.

File: pkgs/development/tools/buf/default.nix [hunk 2]
```diff
@@ -51,6 +54,14 @@ buildGoModule rec {
       cp "$GOPATH/bin/$FILE" "$out/bin/"
     done
 
+    # Completions
+    installShellCompletion --cmd buf --bash <($out/bin/buf bash-completion) \
+      --fish <($out/bin/buf fish-completion) --zsh <($out/bin/buf zsh-completion)
```
Comment (lrewega) [inline pkgs/development/tools/buf/default.nix:59]:
```suggestion
    installShellCompletion --cmd buf \
      --bash <($out/bin/buf bash-completion) \
      --fish <($out/bin/buf fish-completion) \
      --zsh <($out/bin/buf zsh-completion)
```

File: pkgs/development/tools/buf/default.nix [hunk 3]
```diff
@@ -51,6 +54,14 @@ buildGoModule rec {
...
+
+    # Man Pages
+    mkdir man && $GOPATH/bin/buf manpages man
```
Comment (lrewega) [inline pkgs/development/tools/buf/default.nix:62]: We use `$out` for completion above but `$GOPATH` here, I would prefer `$GOPATH` for both.

**Review by lrewega — CHANGES_REQUESTED** [review]

**Review by lrewega — APPROVED** [review]
Looks good, thanks!

**Review by raboof — APPROVED** [review]
Nice!

**Pull request merged**